pkg/utils/pprof.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 PingCAP, Inc. Licensed under Apache-2.0.
+
+package utils
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/pprof"
+	"sync"
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+)
+
+var (
+	startedPProf = ""
+	mu           = new(sync.Mutex)
+)
+
+// StartPProfListener forks a new goroutine listening on specified port and provide pprof info.
+func StartPProfListener(statusAddr string) {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if startedPProf != "" {
+		log.Warn("Try to start pprof when it has been started, nothing will happen", zap.String("address", startedPProf))
+		return
+	}
+
+	go func() {
+		_ = pprof.Handler
```

Comment:
🤔 

let use the proper way make sure pprof is registered:

```go
import ( 
    ...
    // register HTTP handler for /debug/pprof
    _ "net/http/pprof"
)
```

Reply:
The original version of starting pprof uses this. Seems if we use import for side-effect, `gosec` goes unhappy...

`G108: Profiling endpoint is automatically exposed on /debug/pprof (gosec)`

Reply:
Seems [add a `nosec` pragma](https://github.com/securego/gosec/issues/418) would be a solution for this.